sdk/python/tests/components/test_python_op.py

```diff
@@ -621,6 +625,16 @@ def write_to_file_path(number_file: OutputBinaryFile(int)):
         self.helper_test_component_using_local_call(task_factory, arguments={}, expected_output_values={'number_file': '42'})
 
 
+    def test_packages_to_install_feature(self):
+        task_factory = comp.func_to_container_op(dummy_in_0_out_0, packages_to_install=['six', 'pip'])
+
+        self.helper_test_component_using_local_call(task_factory, arguments={}, expected_output_values={})
+
+        task_factory2 = comp.func_to_container_op(dummy_in_0_out_0, packages_to_install=['bad-package-0ee7cf93f396cd5072603dec154425cd53bf1c681c7c7605c60f8faf7799b901'])
```

Comment:
Just out of curiosity, is that possible to shorten the bad package name? 
Another question I have is, does `helper_test_component_using_local_call` actually execute this pip install here? If so, do we have any concern regarding its reliability, because we've seen cases where pip install is temporarily unavailable.

Thanks!

Reply:
>is that possible to shorten the bad package name?

Yes. I've shortened it.

>Another question I have is, does helper_test_component_using_local_call actually execute this pip install here?

Yes the whole command-line is executed.

>If so, do we have any concern regarding its reliability, because we've seen cases where pip install is temporarily unavailable.

We have couple of tests that rely on internet being available (e.g. for `load_component_from_url`). This test also relies on Internet (and PyPI) being available. I do not see a big difference. (I was hesitating about running `pip` as part of tests since it's a bit slow, but I found no better way to test this feature.)
We could later introduce a test switch which allows skipping online-only tests.

Reply:
Agree that it is not something easy to mock. Let's keep it as it is.

Reply:
Forgot to say. I've chosen the packages that are always installed (e.g. the `pip` itself), so this should not download and install any new packages.